CMakeLists.txt

```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.13)
```

Comment:
Can we move this one under src/ and avoid having a CMakeLists.txt in the root of the repo? I'd like this to work kind of like our Windows/VS setup does, where everything is contained in src/.

Reply:
I'm sure it's technically possible, but it would go against the convention, virtually every project based on cmake had this file in tour directory. I'll see if there's any technical advice for going one way or the other, feel free to the same. Let me know how you want this done.

Reply:
It's true that _more_ projects seem to have it in the root, but there is precedent for having it in a first-level subdirectory (see, for example, [libphonenumber](https://github.com/google/libphonenumber/blob/master/cpp/CMakeLists.txt) or [vcpkg](https://github.com/microsoft/vcpkg/blob/master/toolsrc/CMakeLists.txt)).

Given the layout we currently have for this repo, I'd still prefer it in src/.

Reply:
While true, you seem to be skipping over the fact that both those repos (and I assume most others) have a main build _thing_ in the root directory. For libphonenumber, there's `pom.xml` and the library is primarily Java, while the cpp version is a port that could very well be extracted to its own repo, while for vcpkg there're `bootstrap-vcpkg.{bat,sh}` scripts in the root of repository.

Personally, I think it makes sense especially e.g. when you have `docs`, `tests`, `assets`, next to `src` that are also consumed when building. That said, I don't see cmake taking place of the already present visual studio project file and there are only files under`src` being used.

I will make the requested change later today.